vertical-pod-autoscaler/pkg/recommender/input/history/history_provider.go

```diff
@@ -190,8 +190,12 @@ func (p *prometheusHistoryProvider) readLastLabels(res map[model.PodID]*PodHisto
 
 func (p *prometheusHistoryProvider) GetClusterHistory() (map[model.PodID]*PodHistory, error) {
 	res := make(map[model.PodID]*PodHistory)
-	podSelector := fmt.Sprintf("job=\"%s\", %s=~\".+\", %s!=\"POD\", %s!=\"\"",
-		p.config.CadvisorMetricsJobName, p.config.CtrPodNameLabel,
+	var podSelector string
+	if p.config.CadvisorMetricsJobName != "" {
+		podSelector = fmt.Sprintf("job=\"%s\", ", p.config.CadvisorMetricsJobName)
+	}
+	podSelector = fmt.Sprintf("%s%s=~\".+\", %s!=\"POD\", %s!=\"\"",
```

Comment:
Could you change this for podSelector = podSelector+fmt.Sptrintf(%s=~\".+\", %s!=\"POD\", %s!=\"\"", ...)

Reply:
Done